shell/server/proxy.js

```diff
@@ -36,6 +36,85 @@ SANDSTORM_ALTHOME = Meteor.settings && Meteor.settings.home;
 SANDSTORM_LOGDIR = (SANDSTORM_ALTHOME || "") + "/var/log";
 SANDSTORM_VARDIR = (SANDSTORM_ALTHOME || "") + "/var/sandstorm";
 
+var storeReferralProgramInfoApiTokenCreated = function(db, accountId, identityId, apiTokenAccountId) {
+  // From the Referral program's perspective, if Bob's Account has no referredByComplete, then we
+  // update Bob's Identity to say it's referredBy Alice's Account (which is apiTokenAccountId).
+  check(accountId, String);
+  check(identityId, String);
+  check(apiTokenAccountId, String);
+
+  // Bail out early if quota enforcement is disabled.
+  if (! Meteor.settings.public.quotaEnabled) {
+    return;
+  }
+
+  var aliceAccountId = apiTokenAccountId;
+  var bobAccountId = accountId;
+  var bobIdentityId = identityId;
+
+  if (Meteor.users.find({_id: bobAccountId,
```

Comment:
Oh, I misread -- this one is querying the account, while the one below is querying the identity.

That said, I think I would argue that after a referral is complete, you should leave `referredBy` set on the identity, and then in this code here you can do just one query rather than two. It feels easier to reason about that way, IMO, and probably saves queries / database fragmentation overall.